From dafa41a72190c0fa02afe6acdc06f05eb0eda937 Mon Sep 17 00:00:00 2001
From: Gary Lin <glin@suse.com>
Date: Wed, 6 Nov 2019 11:43:44 +0800
Subject: [PATCH] Keep the files in the OTHER directory

We currently only kept the "*.log" files for the repackaging while there
are some use cases that the user might need other types of files.

Update pesign-repackage.spec.in to filter out the meta and internal files
and keep the files in the OTHER directory.

Bugzilla entry:
 OBS do not export some files to API OTHER on x86_64
 https://bugzilla.suse.com/show_bug.cgi?id=1155474

Signed-off-by: Gary Lin <glin@suse.com>
---
 pesign-repackage.spec.in | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/pesign-repackage.spec.in b/pesign-repackage.spec.in
index ca8d325..1679878 100644
--- a/pesign-repackage.spec.in
+++ b/pesign-repackage.spec.in
@@ -84,10 +84,18 @@ for rpm in %_sourcedir/*.rpm; do
 	rpms=("${rpms[@]}" "$rpm")
 done
 popd
-for log in %_sourcedir/*.log; do
-	if test -e "$log"; then
+# Copy files other than the meta files and RPMs to %_topdir/OTHER
+OTHER_FILES=`find %_sourcedir/ -maxdepth 1 -type f \
+	-not -regex '.*\.\(rpm\|spec\|rsasign\|sig\|crt\)' \
+	-not -name "_buildenv" \
+	-not -name "_statistics" \
+	-not -name "logfile" \
+	-not -name "meta" \
+	-print`
+for file in $OTHER_FILES; do
+	if test -e "$file"; then
 		mkdir -p "%_topdir/OTHER"
-		cp "$log" "$_"
+		cp "$file" "$_"
 	fi
 done
 mkdir rsasigned
-- 
2.23.0

